services/submission.js

```diff
@@ -610,6 +610,48 @@ class Submission {
         return ValidationHandler.success(`${res.filter(result => result.status === 'fulfilled').length} extra files deleted`);
     }
 
+    /**
+     * deleteInactiveSubmission
+     * description: overnight job to set inactive submission status to "Deleted", delete related data and files
+     */
+    async deleteInactiveSubmissions(){
+        //get target inactive date, current date - config.inactive_submission_days (default 120 days)
+        var target_inactive_date = new Date();
+        target_inactive_date.setDate(target_inactive_date.getDate() - config.inactive_submission_days);
+        const query = [{"$match": {"status": IN_PROGRESS, "accessedAt": {"$exists": true, "$ne": null, "$lte": target_inactive_date}}}];
+        try {
+            const inactive_subs = await this.submissionCollection.aggregate(query);
+            if (!inactive_subs || inactive_subs.length === 0) {
+                console.debug("No inactive submission found.")
+                return "No inactive submissions";
+            }
+            //delete files under inactive submissions
+            const promises = Array.from(inactive_subs).map(aSub => this.s3Service.deleteDirectory(aSub.bucketName, aSub.rootPath));
```

Comment:
It expects promise from deleteDirectory, but that function seems doesn't return anything.

Reply:
updated codes not to expect return a promise but wait completion of deletion.   The deleteDirectiroy function is too complicated to implement nested promise since it call itself for deleting dir itself after all objects under dir are deleted.